doc/userguide/lua/libs/flowvar.rst

```diff
@@ -0,0 +1,79 @@
+Flowvar
+#######
+
+The ``suricata.flowvar`` library exposes flow variables to Lua
+scripts.
+
+Initialization
+--------------
+
+First, the ``flowvar`` lib must be required::
+
+  local flowvarlib = require("suricata.flow")
+
+Then in the ``init`` method, any flow variables used in the script
+should be registered. This is optional and could be skipped if you
+know for sure the flow variable will be registered by some other
+means.
```

Comment:
does this mean that the changes in this commit fix the issue that the commit described, then? (just to confirm that I understood things correctly)
Or should we add a note about that somewhere?

Reply:
No, this is documenting the new behavior.  In current git master, the "needs" syntax was used which was the only way to register a flowvar, and always happened during the init phase. No reference to old behavior is made here.